src/test/cdk/detectCdkProjects.test.ts

```diff
@@ -0,0 +1,194 @@
+/*!
+ * Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as del from 'del'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { detectCdkProjects, detectLocalCdkProjects } from '../../cdk/explorer/detectCdkProjects'
+import * as filesystem from '../../shared/filesystem'
+import { createWorkspaceFolder } from './util'
+
+function normalizePath(...paths: string[]): string {
+    return vscode.Uri.file(path.join(...paths)).fsPath
+}
+
+describe('detectCdkProjects', () => {
+    const workspacePaths: string[] = []
+    const workspaceFolders: vscode.WorkspaceFolder[] = []
+
+    beforeEach(async () => {
+        const { workspacePath, workspaceFolder } = await createWorkspaceFolder('vsctk-shiv')
+
+        workspacePaths.push(workspacePath)
+        workspaceFolders.push(workspaceFolder)
+    })
+
+    afterEach(async () => {
+        await del(workspacePaths, { force: true })
+
+        workspacePaths.length = 0
+        workspaceFolders.length = 0
+    })
+
+    it('detects no projects when workspaceFolders is undefined', async () => {
+        const actual = await detectCdkProjects(undefined)
+
+        assert.ok(actual)
+        assert.strictEqual(actual.length, 0)
+    })
+
+    it('detects no projects when workspaceFolders is empty', async () => {
+        const actual = await detectCdkProjects([])
+
+        assert.ok(actual)
+        assert.strictEqual(actual.length, 0)
+    })
+
+    it('detects no projects when tree.json does not exist', async () => {
+        const actual = await detectCdkProjects(workspaceFolders)
+
+        assert.ok(actual)
+        assert.strictEqual(actual.length, 0)
+    })
+
+    it('Detects projects at the root of each workspace folder', async () => {
```

Comment:
Tests like this (for the rest of this file), its acceptable to write files into these temporary locations to avoid mocking simple file operations. Keeps the tests readable too :)

Reply:
🙏 thank you! I was reusing this to stay aligned with how the SAM Local stuff is set up. Modified tests and removed these.